src/Network_Equivalences-Communication.thy

```diff
@@ -13,6 +13,33 @@ lemma adapted_after_distributor:
   shows "(A \<Rightarrow> Bs) \<guillemotleft> \<E> = A \<guillemotleft> \<E> \<Rightarrow> map (\<lambda>B. B \<guillemotleft> \<E>) Bs"
   sorry
 
+lemma transition_from_distributor:
+  assumes "A \<Rightarrow> Bs \<rightarrow>\<^sub>s\<lparr>\<alpha>\<rparr> Q"
+  obtains n and X
+  where
+    "\<alpha> = A \<triangleright> \<star>\<^bsup>n\<^esup> X"
+  and
+    "Q = \<Prod>B \<leftarrow> Bs. B \<guillemotleft> suffix n \<triangleleft> X \<parallel> A \<guillemotleft> suffix n \<Rightarrow> map (\<lambda>B. B \<guillemotleft> suffix n) Bs"
```

Comment:
Our current practice is that `obtain` commands that don’t fit on a single line have their `where` clauses indented (see in particular the Þ-calculus formalization). We should apply the same rule also in the case of `obtains` and consequently indent this whole `where` clause.

Reply:
Apparently, you haven’t adapted the statement of this lemma to the middle-ground scheme. The middle-ground scheme would mandate to state that `Q` is equal to `∏B ← Bs. B « suffix n ◃ X ∥ (A ⇒ Bs) « suffix n`, as for the repetition of the original process family, `« suffix n` should be on the outside.

Reply:
Fixed in f703d9e471c70cd3db997e0de6ab1ad67e1e66de, please verify.